docs/development/design/exit-node.md

```diff
@@ -0,0 +1,145 @@
+    - DNS UDP 53. This should like be discovered via host discovery. There could potentially be scenarios where a DNS is only available via the Nexd mesh but that would be fairly atypical and should be tackled with Nexodus DNS implementation. This rule could also be narrowed if there was a host entry added to the device the agent is running on.
+    - STUN requests, UDP 3478 and 19302 for the RFC defined and Google STUN ports. This is required because once a device is forwarding all traffic through the exit node, the reflexive address for the client device would take on the same reflexive address of the exit node.
+2. To enable forwarding if the driver interface is down, A routing table with a default route to the driver interface such as wg0 is created. A forwarding mark is added to the driver of `51820` to denote Wireguard that says, if a packet does not have the firewall mark 51820, perform a lookup in the routing table 51820. Inside of that routing table 51820 is a default route pointing to wg0.
+
+A diagram of the OOB services is as follows:
+
+```mermaid
+graph LR
+subgraph Nexodus Mesh with All Devices Peered to one another
+Nex_Device_4["Nexd-1 (100.100.0.1)
+exit-node client"]
+Nex_Device_5["Nexd-2 (100.100.0.2)
+exit-node client"]
+Nex_Device_6["Nexd-3 (100.100.0.3)
+exit-node client"]
+Nexodus_Network_Router_B["Nexd-4 Exit Node 
+wg0 100.100.0.4
+eth(n) 172.16.100.10/24"]
+end
+
+subgraph External Network 
+Internet["Network or Internet
+Receives Traffic from
+the Nexodus Network
+as sourced from eth(n)
+172.16.100.10 via SNAT"]
+end
+
+Nexodus_Network_Router_B-- "eth(n)" --> Internet
+Nex_Device_4 -- wg0 --> Nexodus_Network_Router_B
+Nex_Device_5 -- wg0 --> Nexodus_Network_Router_B
+Nex_Device_6 -- wg0 --> Nexodus_Network_Router_B
+
+
+subgraph Nexodus Out of Band Services
+DNS["DNS UDP Port 53"]
+API["API-Server TCP 443"]
+STUN["STUN Servers UDP [3478,19302]"]
+end
+
+Nex_Device_4 -- "eth0(n)" --> STUN
+Nex_Device_4 -- "eth0(n)" --> DNS
+Nex_Device_4 -- "eth0(n)" --> API
+```
+
+## Alternatives Considered
+
+- We could take a namespace approach but that requires moving physical interfaces into namespaces which can be pretty invasive on a machine. The proposed approach makes teardown as simple as removing a netfilter table and two routing tables with a few policy routing rules.
+- For Darwin support we will need to take a path of granular routes which is more invasive and less flexible for out of band services that we can achieve with native Linux routing capabilities, but it could be considered an alternative option. It will require some Darwin networking research to try and send the stun through a policy based route. An exit node being Linux only seems reasonable for now. Below is an example:
```

Comment:
+1 for linux-only support for exit nodes. That seems very reasonable to me.

Reply:
```suggestion
- For Darwin support, we will need to take a path of granular routes which is more invasive and less flexible for out-of-band services than we can achieve with native Linux routing capabilities, but it could be considered an alternative option. It will require some Darwin networking research to try and send the stun through a policy-based route. An exit node being Linux only seems reasonable for now. Below is an example:
```